astropy/utils/console.py

```diff
@@ -631,10 +637,17 @@ def _update_ipython_widget(self, value=None):
         # if none exists.
         if not hasattr(self, '_widget'):
             # Import only if an IPython widget, i.e., widget in iPython NB
-            from IPython.html import widgets
+            try:
+                from ipywidgets import widgets
+            except ImportError:
+                from IPython.html import widgets
             from IPython.display import display
 
-            self._widget = widgets.FloatProgressWidget()
+            try:
+                self._widget = widgets.FloatProgress()
+            except AttributeError:
+                self._widget = widgets.FloatProgressWidget()
```

Comment:
The problem is that `ipywidgets` does not contain the old IPython 2 widget shims anymore. I'm dealing with this transition over in https://github.com/mwcraig/reducer/tree/ipython-3

There I'm not supporting IPython 2. What I'd recommend here if we want to support ipython 2 is checking at the top for ipython version, then for 2 only `from IPython.html.widgets import FloatProgressWidget as FloatProgress` and otherwise import `FloatProgress` from the appropriate place for v3 or v4.